### PR TITLE
add homework4 11.12.23

### PR DESCRIPTION
**Pull request by olemykha**
added homework4 11.12.23
all conditions are met. no errors from the linter.

### CODE REVIEW

File: homework/homework111223.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+"""HOMEWORK4."""
```
Comment (antohaUa) [inline homework/homework111223.py]: Please add task description into dostring.

File: homework/homework111223.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Continuous as the stars that shine
+And twinkle on the Milky Way,
+They stretched in never-ending line
+Along the margin of a bay:
+Ten thousand saw I at a glance,
+Tossing their heads in sprightly dance.
+"""
+
+poem_text.lower()
```
Comment (antohaUa) [inline homework/homework111223.py]: Here you have real bug - result of this operation should be saved 
```suggestion
poem_text = poem_text.lower()
```

File: homework/homework111223.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# Counting the number of vowels in a text
+for vowel in vowels:
+    count = poem_text.count(vowel)
+    logging.info(f'| {vowel:^6}|  {count:^5} |')
+
+logging.info('-' * separator_number)
+logging.info('The table output is complete.')
+
+# The second task is to change all the vowels to those provided by the teacher.
+modified_poem_text = (
```
Comment (antohaUa) [inline homework/homework111223.py]: As advanced try to make the same but with maketarns and translate methods

**Review by antohaUa — CHANGES_REQUESTED** [review]
Bug detected )

File: homework/homework111223.py [hunk 4]
```diff
@@ -23,29 +41,29 @@
...
-)
+# Task 2. Changing all the vowels to those provided by the teacher.
+new_symbols_for_replacement = str.maketrans({
+    'A': 'À', 'a': 'à',
+    'E': 'É', 'e': 'é',
+    'I': 'Í', 'i': 'í',
+    'O': 'Ó', 'o': 'ó',
+    'U': 'Ú', 'u': 'ú',
+})
+modified_poem_text = poem_text.translate(new_symbols_for_replacement)
```
Comment (antohaUa) [inline homework/homework111223.py]: Final poem ( replaced text)  in you case will be without capital letters. Try to understand where in your code fix required

**Review by antohaUa — APPROVED** [review]
Good

File: homework/homework111223.py (hunk 4 above)
Comment (olemykha) [reply homework/homework111223.py]: Yeah, you're right. I see

File: homework/homework111223.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+Along the margin of a bay:
+Ten thousand saw I at a glance,
+Tossing their heads in sprightly dance.
+"""
+
+poem_text = initial_poem_text.lower()
+vowels = ['a', 'e', 'i', 'o', 'u']  # list of vowels
+
+separator_number = 18
+logging.info('-' * separator_number)
```
Comment (antohaUa) [inline homework/homework111223.py:48]: The best practice for such case create list object and then append lines.
Aka 
```
table =  []
table.append('-' * separator_number)
...
```
and then just `logging.info(table)` at the end.

But all above related to advanced level )

**Review by antohaUa — APPROVED** [review]
Good

**Pull request merged**